File: test_data/code/python/1496-path-crossing.py
Python
class Solution:
    def isPathCrossing(self, path: str) -> bool:
        x, y = 0, 0
        visited = set((0, 0))
        for dir in path:
            if dir == 'N':
                y += 1
            elif dir == 'S':
                y -= 1
            elif dir == 'E':
                x += 1
            elif dir == 'W':
                x -= 1
            if (x, y) in visited:
                return True
            visited.add((x, y))
        return False

File: test_data/code/java/1496-path-crossing.java
class Solution {
    public boolean isPathCrossing(String path) {
        int x = 0, y = 0;
        boolean[][] visited = new boolean[2001][2001];
        visited[1001][1001] = true;
        for (char c : path.toCharArray()) {
            switch (c) {
                case 'N':
                    y++;
                    break;
                case 'S':
                    y--;
                    break;
                case 'E':
                    x++;
                    break;
                case 'W':
                    x--;
                    break;
            }
            if (x < 0 || x >= 2000 || y < 0 || y >= 2000 || visited[x + 1000][y + 1000]) {
                return true;
            }
            visited[x + 1000][y + 1000] = true;
        }
        return false;
    }
}